ts/components/services/ContactPreferencesToggles/__test__/PreferenceToggleRow.test.tsx

```diff
@@ -35,17 +35,67 @@ describe("PreferenceToggleRow component", () => {
       expect(component.getByTestId("new-test-id")).toBeDefined();
     });
   });
+
+  describe("handle different status", () => {
+    it("should display activity indicator", () => {
+      const component = renderComponent({
+        ...options,
+        isError: false,
+        isLoading: true
+      });
+
+      expect(
+        component.getByTestId("preference-toggle-row-loading")
+      ).toBeDefined();
+      expect(component.toJSON()).toMatchSnapshot();
```

Comment:
I see three more snapshots here.
The one I added [above](https://github.com/pagopa/io-app/pull/3186/files#diff-5d553648ffded26efaea86a92d1eeade5f7a1a2287b1a021f8f2def22af41e16R14) is meant to guarantee  that the DOM structure and CSS remain stable through development.

Being verbose and difficult to diff by nature, I strongly suggest the following:
1. don't mix UI queries and spies with snapshots (to test one thing at a time)
2. use snapshots sparingly to improve the signal-to-noise ratio and speed up reviews

It might be worth roping in our colleagues in this discussion.

Reply:
it was a test, since the component will appear completely different in case of error or loading state. Maybe we can discuss it better, if you prefer we can clean up these test cases here, what do you think?